### PR TITLE
fix: add misc fixes before `_comp_compgen` changes

### PR DESCRIPTION
**Pull request by akinomyoga**

### CODE REVIEW

File: completions/iptables [hunk 1]
```diff
@@ -5,33 +5,34 @@ _comp_cmd_iptables()
...
                 COMPREPLY=($(compgen -W 'ACCEPT DROP LOG ULOG REJECT
-                `"$1" $table -nL 2>/dev/null | command sed -ne "$chain" \
+                `"$1" ${table:+-t "$table"} -nL 2>/dev/null | command sed -ne "$chain" \
                 -e "s/INPUT|OUTPUT|FORWARD|PREROUTING|POSTROUTING//"`' -- \
                     "$cur"))
-            elif [[ $table == "-t nat" ]]; then
+            elif [[ $table == "nat" ]]; then
                 COMPREPLY=($(compgen -W 'ACCEPT DROP LOG ULOG REJECT MIRROR SNAT
-                DNAT MASQUERADE `"$1" $table -nL 2>/dev/null | \
+                DNAT MASQUERADE `"$1" ${table:+-t "$table"} -nL 2>/dev/null | \
```
Comment (scop) [inline completions/iptables:30]: `[[ $table == "nat" ]]` checked above.
```suggestion
                DNAT MASQUERADE `"$1" -t "$table" -nL 2>/dev/null | \
```

File: completions/iptables [hunk 2]
```diff
@@ -5,33 +5,34 @@ _comp_cmd_iptables()
...
                 command sed -ne "$chain" -e "s/OUTPUT|PREROUTING|POSTROUTING//"`' \
                     -- "$cur"))
-            elif [[ $table == "-t mangle" ]]; then
+            elif [[ $table == "mangle" ]]; then
                 COMPREPLY=($(compgen -W 'ACCEPT DROP LOG ULOG REJECT MARK TOS
-                `"$1" $table -nL 2>/dev/null | command sed -ne "$chain" \
+                `"$1" ${table:+-t "$table"} -nL 2>/dev/null | command sed -ne "$chain" \
```
Comment (scop) [inline completions/iptables:35]:
```suggestion
                `"$1" -t "$table" -nL 2>/dev/null | command sed -ne "$chain" \
```

File: completions/ebtables [hunk 3]
```diff
@@ -5,38 +5,39 @@ _comp_cmd_ebtables()
...
-                $("$1" $table -L 2>/dev/null | \
+                $("$1" ${table:+-t "$table"} -L 2>/dev/null | \
                   command sed -n -e "s/INPUT\|OUTPUT\|FORWARD//" \
                                  -e "$chain")' \
                     -- "$cur"))
-            elif [[ $table == "-t nat" ]]; then
+            elif [[ $table == "nat" ]]; then
                 COMPREPLY=($(compgen -W '$targets
-                $("$1" $table -L 2>/dev/null | \
+                $("$1" ${table:+-t "$table"} -L 2>/dev/null | \
```
Comment (scop) [inline completions/ebtables:34]: `[[ $table == "nat" ]]` checked above.
```suggestion
                $("$1" -t "$table" -L 2>/dev/null | \
```

File: completions/ebtables [hunk 4]
```diff
@@ -5,38 +5,39 @@ _comp_cmd_ebtables()
...
                   command sed -n -e "s/OUTPUT|PREROUTING|POSTROUTING//" \
                                  -e "$chain")' \
                     -- "$cur"))
-            elif [[ $table == "-t broute" ]]; then
+            elif [[ $table == "broute" ]]; then
                 COMPREPLY=($(compgen -W 'ACCEPT DROP
-                $("$1" $table -L 2>/dev/null | \
+                $("$1" ${table:+-t "$table"} -L 2>/dev/null | \
```
Comment (scop) [inline completions/ebtables:40]:
```suggestion
                $("$1" -t "$table" -L 2>/dev/null | \
```

**Review by scop — APPROVED** [review]
Pre-approved with some suggestions.

File: completions/iptables (hunk 1 above)
Comment (akinomyoga) [reply completions/iptables:30]: Ah, right!

**Pull request merged**